data/src/main/java/com/example/data/repository/ApolloCountryRepositoryImpl.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.data.repository
+
+import com.apollographql.apollo3.ApolloClient
+import com.example.data.CountriesQuery
+import com.example.data.mapper.toPost
+import com.example.domain.model.Post
+import com.example.domain.repository.ApolloCountryRepository
+
+class ApolloCountryRepositoryImpl(private val apolloClient: ApolloClient) :
+    ApolloCountryRepository {
+
+    override suspend fun getCountries(): List<Post> {
+        return try {
+            apolloClient.query(CountriesQuery())
+                .execute()
+                .data
+                ?.countries.orEmpty()
+                .map { it.toPost() }
+        } catch (e: Exception) {
+            e.printStackTrace()
+            emptyList()
+        }
```

Comment:
The `try-catch` block should be a part of the usecase. Repository should only be responsible for making data-source calls (network/local) and `throw`ing appropriate Exceptions based on data source's response, usecase would be the one to handle all such Exceptions.